src/devices/Media/VideoDevice/Devices/UnixVideoDevice.cs

```diff
@@ -0,0 +1,546 @@
+        {
+            await Task.Run(() =>
+            {
+                Initialize();
+                SetVideoConnectionSettings();
+                byte[] dataBuffer = ProcessCaptureData();
+                Close();
+
+                using (FileStream fs = new FileStream(path, FileMode.Create))
+                {
+                    fs.Write(dataBuffer, 0, dataBuffer.Length);
+                    fs.Flush();
+                }    
+            }, token);
+        }
+
+        /// <summary>
+        /// Capture a picture from the video device.
+        /// </summary>
+        /// <returns>Picture stream</returns>
+        public override async Task<MemoryStream> CaptureAsync()
+        {
+            using (CancellationTokenSource cts = new CancellationTokenSource())
+            {
+                return  await CaptureAsync(cts.Token);
+            }
+        }
+
+        /// <summary>
+        /// Capture a picture from the video device.
+        /// </summary>
+        /// <returns>Picture stream</returns>
+        /// <param name="token">A cancellation token that can be used to cancel the work</param>
+        public override async Task<MemoryStream> CaptureAsync(CancellationToken token)
+        {
+            return await Task.Run(() =>
+            {
+                Initialize();
+                SetVideoConnectionSettings();
+                byte[] dataBuffer = ProcessCaptureData();
+                Close();
+
+                return new MemoryStream(dataBuffer);
+            }, token);
+        }
+
+        /// <summary>
+        /// Query controls value from the video device.
+        /// </summary>
+        /// <param name="type">The type of a video device's control.</param>
+        /// <returns>The default and current values of a video device's control.</returns>
+        public override VideoDeviceValue GetVideoDeviceValue(VideoDeviceValueType type)
+        {
+            // Get default value
+            v4l2_queryctrl query = new v4l2_queryctrl
+            {
+                id = type
+            };
+            V4l2Struct(VideoSettings.VIDIOC_QUERYCTRL, ref query);
+
+            // Get current value
+            v4l2_control ctrl = new v4l2_control
+            {
+                id = type,
+            };
+            V4l2Struct(VideoSettings.VIDIOC_G_CTRL, ref ctrl);
+
+            return new VideoDeviceValue
+            {
+                Name = type.ToString(),
+                Minimum = query.minimum,
+                Maximum = query.maximum,
+                Step = query.step,
+                DefaultValue = query.default_value,
+                CurrentValue = ctrl.value
+            };
+        }
+
+        /// <summary>
+        /// Get all the pixel formats supported by the device.
+        /// </summary>
+        /// <returns>Supported pixel formats</returns>
+        public override List<PixelFormat> GetSupportedPixelFormats()
```

Comment:
this should return `IEnumerable<PixelFormat>`. List is fine as an implementation detail though